onset_maproom/maproom.py

```diff
@@ -255,6 +257,18 @@ def map_click(click_lat_lng):
     ), round(lat_lng[0],4), round(lat_lng[1],4)
 
 
+@APP.callback(
+    Output("hover_feature_label", "children"),
+    Input({"type": "borders_adm", "index": ALL}, "hover_feature")
+)
+def write_hover_adm_label(adm_loc):
+    location_description = "the map will return location name"
+    for i, adm in enumerate(adm_loc):
```

Comment:
Only the last (topmost) input is used, so there's no point using a loop. Just use `adm_loc[-1]` (unless `len(adm_loc) == 0` in which case we use the default).

Reply:
So for this input it is not using all three of the layers to construct the output, it is only using a single adm layer? What determines the order of inputs that you know you could always use adm_loc[-1]? Are the layers always ordered in a certain way?

Reply:
From what I observed, let's say there are 3 admin boundaries, ie 3 components with id's type = borders_adm, hover_feature returns always an array of 3. However, it does the job of returning the feature only for those that are checked, and returns None otherwise. So there is no guarantee the last would not be None and so we want the last (because their index goes from 1 to 3 as I set it up with adm_lvl) that is not None.

The only flaw in the current implementation is that if the user unchecks all the admin boxes, then mousing over the map would return "Mousing over the map will return location name" even when the map is somewhere over Ethiopia...

Reply:
For instance in Ethiopia we have 3 levels, by default level 1 and 2 are checked, so hover_feature returns:
[feature_1, feature_2, None]

Reply:
I see.